src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -508,20 +509,29 @@ public CaseWorkerProfile mapCaseWorkerProfileRequest(String idamId,
     // Idam_UP call.
     @SuppressWarnings("unchecked")
     private ResponseEntity<Object> createUserProfileInIdamUP(CaseWorkersProfileCreationRequest cwrdProfileRequest) {
-
+        ResponseEntity<Object> responseEntity;
         Response response = null;
         Object clazz;
         try {
             long time1 = System.currentTimeMillis();
             response = userProfileFeignClient.createUserProfile(createUserProfileRequest(cwrdProfileRequest));
             log.info("{}:: Time taken to call UP is {}", loggingComponentName, (System.currentTimeMillis() - time1));
-            if (response.status() == 201 || response.status() == 409) {
-                clazz = UserProfileCreationResponse.class;
-            } else {
-                clazz = ErrorResponse.class;
-                validationServiceFacade.logFailures(UP_CREATION_FAILED, cwrdProfileRequest.getRowId());
+
+            clazz = (response.status() == 201 || response.status() == 409)
+                    ? UserProfileCreationResponse.class : ErrorResponse.class;
+
+            responseEntity = JsonFeignResponseUtil.toResponseEntity(response, clazz);
+            if (clazz == ErrorResponse.class && nonNull(responseEntity.getBody())) {
```

Comment:
I think you need to write unit test cases for these changes - https://sonarcloud.io/component_measures?id=uk.gov.hmcts.reform%3Areform%3A%3Ard-caseworker-ref-api&metric=new_coverage&pullRequest=230&selected=uk.gov.hmcts.reform%3Areform%3A%3Ard-caseworker-ref-api%3Asrc%2Fmain%2Fjava%2Fuk%2Fgov%2Fhmcts%2Freform%2Fcwrdapi%2Fservice%2Fimpl%2FCaseWorkerServiceImpl.java&view=list

Reply:
Added the test case, which bumps up the coverage.

Reply:
Thanks much @nitishsingla1234 . Coverage has been improved drastically :)